extensions/vscode/src/views/lessonsTree.ts

```diff
@@ -0,0 +1,164 @@
+
+  constructor(
+    private readonly _workspaceRoot: vscode.Uri,
+    private _context: vscode.ExtensionContext,
+  ) {
+    this._loadLessons();
+  }
+
+  private _loadLessons(): void {
+    try {
+      const tutorialFolderPath = vscode.Uri.joinPath(this._workspaceRoot, 'src', 'content', 'tutorial').fsPath;
+      this._isTutorialKitWorkspace = true;
+      this._lessons = this._loadLessonsFromFolder(tutorialFolderPath);
+    } catch {
+      this._isTutorialKitWorkspace = false;
+    }
+  }
+
+  private _loadLessonsFromFolder(folderPath: string): Lesson[] {
+    const lessons: Lesson[] = [];
+    const files = fs.readdirSync(folderPath);
+
+    for (const file of files) {
+      const filePath = path.join(folderPath, file);
+      const stats = fs.statSync(filePath);
+
+      if (stats.isDirectory()) {
+        const lessonName = path.basename(filePath);
+        const subLessons = this._loadLessonsFromFolder(filePath);
+        const lesson = new Lesson(lessonName, filePath, subLessons);
+
+        // check if the folder directly includes one of the metadata files
+        const folderFiles = fs.readdirSync(filePath);
+        const metadataFile = folderFiles.find((folderFile) => metadataFiles.includes(folderFile));
+
+        if (metadataFile) {
+          const metadataFilePath = path.join(filePath, metadataFile);
+          const metadataFileContent = fs.readFileSync(metadataFilePath, 'utf8');
+          const parsedContent = grayMatter(metadataFileContent);
+          lesson.name = parsedContent.data.title;
+          lesson.metadata = {
+            _path: metadataFilePath,
+            ...(parsedContent.data as any),
+          };
+          lessons.push(lesson);
+        }
+      }
+    }
+
+    return lessons;
+  }
+
+  private _onDidChangeTreeData: vscode.EventEmitter<Lesson | undefined> = new vscode.EventEmitter<Lesson | undefined>();
+  readonly onDidChangeTreeData: vscode.Event<Lesson | undefined> = this._onDidChangeTreeData.event;
+
+  refresh(): void {
+    this._loadLessons();
+    this._onDidChangeTreeData.fire(undefined);
+  }
+
+  getTreeItem(lesson: Lesson): vscode.TreeItem {
+    const treeItem = new vscode.TreeItem(lesson.name);
+    treeItem.collapsibleState =
+      lesson.children.length > 0 ? vscode.TreeItemCollapsibleState.Collapsed : vscode.TreeItemCollapsibleState.None;
+
+    treeItem.contextValue = lesson.metadata?.type;
+
+    const shouldOpenFile = lesson.metadata?.type === 'lesson';
+    treeItem.command = {
+      command: cmd.goto.command,
+      title: 'Go to the lesson',
+      arguments: [lesson.path, lesson.metadata, shouldOpenFile],
+    };
+
+    treeItem.iconPath =
+      lesson.metadata?.type === 'lesson' ? getIcon(this._context, 'lesson.svg') : getIcon(this._context, 'chapter.svg');
+
+    return treeItem;
+  }
+
+  getChildren(element?: Lesson): Lesson[] {
+    if (element) {
+      return element.children;
+    }
+
+    return this._lessons;
+  }
+}
+
+export async function useLessonTree() {
```

Comment:
What about all these comments? Can we remove them or is there a specific reason for us to keep them?

Reply:
To have them in the git history. They're removed in #142